bundle/edu.gemini.ags/src/main/scala/edu/gemini/ags/api/AgsHash.scala

```diff
@@ -0,0 +1,148 @@
+package edu.gemini.ags.api
+
+import edu.gemini.shared.util.immutable.ScalaConverters._
+import edu.gemini.shared.util.immutable.{Option => GemOption}
+import edu.gemini.spModel.ags.AgsStrategyKey
+import edu.gemini.spModel.gemini.flamingos2.Flamingos2
+import edu.gemini.spModel.gemini.gmos.{InstGmosNorth, InstGmosSouth}
+import edu.gemini.spModel.obs.context.ObsContext
+
+import java.time.Instant
+
+import scala.collection.mutable.ListBuffer
+import scala.util.hashing.{MurmurHash3 => M3}
+
+import scala.collection.JavaConverters._
+
+/** Computes an `Int` hash value that corresponds to the set of inputs to the
+  * AGS lookup.  If an observation is changed in some way and yet the hash
+  * algorithm computes the same value, there is no need to re-do the AGS search.
+  */
+object AgsHash {
+
+  /** Calculates the AGS hash for this context, using the current base position
+    * location.
+    *
+    * <em>Warning.</em> This method is not referentially transparent as it may
+    * provide distinct answers for the same input at different times
+    * (particularly for non-sidereal targets).
+    */
+  def hashNow(ctx: ObsContext): Int =
```

Comment:
The problem is that the base position is an extremely important factor, and the where the base position is depends on the time.